### DISCUSSION TITLE
Error when launching booth

### DISCUSSION THREAD

**Discussion by lyazide** [photobooth-app/photobooth-app#30] [Q&A] [answered]
Hi,
I made an update and now I get the following error. I tried to recover but do not want to break everything as I have a wedding this weekend.

mai 02 16:09:57 raspberrypi python[1343]:   File "/home/yaz/.local/lib/python3.9                                                                                                                /site-packages/picamera2/encoders/encoder.py", line 6, in <module>
mai 02 16:09:57 raspberrypi python[1343]:     from v4l2 import *
mai 02 16:09:57 raspberrypi python[1343]:   File "/home/yaz/.local/lib/python3.9                                                                                                                /site-packages/v4l2.py", line 197, in <module>
mai 02 16:09:57 raspberrypi python[1343]:     ) = range(1, 9) + [0x80]
mai 02 16:09:57 raspberrypi python[1343]: TypeError: unsupported operand type(s)                                                                                                                 for +: 'range' and 'list'

Any Idea on what is causing this issue?

**Comment by mgineer85** [comment, marked as answer]
Please check whether picamera2 ist installed via pip. 
` pip freeze
or
pip list`
If installed please uninstall as its likely an old version.
Picamera2 is now installed via apt, see picamera2 documentation chapter installation:
https://github.com/raspberrypi/picamera2

**Reply by mgineer85** [reply]
I could add a warning in the installer later if pip package is detected

**Reply by mgineer85** [reply]
If apt and pip is installed both at same time, picamera2 from pip is preferred, but if not updated to most current version it might break with other dependency

**Reply by lyazide** [reply]
Right that was a confict from pip and apt. Now with the last update I do not have the autofocus. If I put the auto it focus but get out of focus when taking a picture. With no focus it do not do autofocus.

**Reply by mgineer85** [reply]
Which camera was it that you use? Camera module 3, right? If so Pls choose in backends settings the continuous autofocus module. If different you might better use interval autofocus module.

**Reply by lyazide** [reply]
I use the Arducam 64Mp.

**Reply by mgineer85** [reply]
I see, that one has no Pdaf algorithm so you need interval autofocus mode. Every 10 seconds and on countdown start autofocus is triggered. If it is blurry, you might need a longer countdown, because focus did not finish yet while taking a picture.

**Reply by lyazide** [reply]
Well. I will play with that. Just by curiosity do you use a canon CP1xxx printer. I try to understand how to print without margin.

**Reply by mgineer85** [reply]
Also I recommend to use only 16mp resolution, the 64mp has too low number of frames for proper autofocus. In general I recommend the imx519 over 64mp camera because it works better with low lighting.

**Reply by mgineer85** [reply]
> Well. I will play with that. Just by curiosity do you use a canon CP1xxx printer. I try to understand how to print without margin.

Did not use a printer on my booth yet, but have cp1300 available for later to use.

**Reply by lyazide** [reply]
The printing time is taking lot of time. So I try to lower the resolution before printing. My main problem at the moment  is I can not get full format printing : always have margins :(

**Reply by mgineer85** [reply]
I think due to different aspect ratio of cam and printer you have either white margin or need to crop the image. Unfortunately, I do not have a ready to use command for you, but if you find it please post somewhere here 😀
